### ISSUE TITLE
[$1000] Web - Profile - The text on the modal Drag, zoom, and rotate your image to your preferred  is highlighted when rotate the picture the picture

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#14139]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

HOLD on https://github.com/Expensify/App/issues/16660

**Issue found when executing PR** https://github.com/Expensify/App/pull/14115

## Action Performed:
1. Login to any account
2. Open the settings page
3. Go to the profile page by clicking on the profile picture or the account email
4. Click on the profile picture and click on the upload photo button
5. Select an image from your device and the upload photo modal will open
6. Select the text on the modal Drag, zoom, and rotate your image to your preferred specifications by double click the text
7. Start Rotate the picture

## Expected Result:
The text on the modal Drag, zoom, and rotate your image to your preferred  is not highlighted when rotate the picture 

## Actual Result:
The text on the modal Drag, zoom, and rotate your image to your preferred  is highlighted when rotate the picture 

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [x] Web / Chrome

**Version Number:** 1.2.50.14

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/211363721-1ae8ed23-a28e-4ad6-8ec0-eff5970dad95.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012de3dfcd27f5f324</li>
        <li>Upwork Job ID: 1613145679138578432</li>
        <li>Last Price Increase: 2023-03-27</li>
    </ul>
</details>

**Comment by twisterdotcom** [comment]
My first few of these, so taking the template from StackOverflowTeams: https://stackoverflowteams.com/c/expensify/questions/14418

- [ ] The "bug" is actually a bug.
- [x] The bug is not a duplicate report of an existing GH
- [x] The bug is reproducible, following the reproduction steps.
- [x] The GH template is filled out as fully as possible
- [x] The GH was created by an Expensify employee or a QA tester.
- [x] The bug is an /App issue

---

I don't really understand this issue. If I highlight some text, and then click somewhere else in the app, the text **should** be unhighlighted shouldn't it? The focus has moved, in this case, to the rotate option.

**Comment by twisterdotcom** [comment]
Ah wait, I see now. It just randomly highlights it whenever you click rotate:

https://user-images.githubusercontent.com/9133401/211802373-ce529da4-6c36-4870-801b-8cca1ea7076c.mp4

https://expensify.slack.com/archives/C01GTK53T8Q/p1673438611054609 - making this `External`

**Comment by tienifr** [comment]
# Proposal

## Problem
The clicking action was triggered for the wrapper div component, causing the text in the div to be highlighted

## Solution

Disable user select in the text and the menu

```diff
diff --git a/src/components/AvatarCropModal/AvatarCropModal.js b/src/components/AvatarCropModal/AvatarCropModal.js
index 1d9922ead4..a024e3b079 100644
--- a/src/components/AvatarCropModal/AvatarCropModal.js
+++ b/src/components/AvatarCropModal/AvatarCropModal.js
@@ -316,7 +316,7 @@ const AvatarCropModal = (props) => {
                 title={props.translate('avatarCropModal.title')}
                 onCloseButtonPress={props.onClose}
             />
-            <Text style={[styles.mh5]}>{props.translate('avatarCropModal.description')}</Text>
+            <Text style={[styles.mh5, styles.userSelectNone]}>{props.translate('avatarCropModal.description')}</Text>
             <GestureHandlerRootView onLayout={initializeImageContainer} style={[styles.alignSelfStretch, styles.m5, styles.flex1, styles.alignItemsCenter]}>
 
                 {/* To avoid layout shift we should hide this component until the image container & image is initialized */}
diff --git a/src/components/HeaderWithCloseButton.js b/src/components/HeaderWithCloseButton.js
index ee1970bca3..9df1cbb204 100755
--- a/src/components/HeaderWithCloseButton.js
+++ b/src/components/HeaderWithCloseButton.js
@@ -136,6 +136,7 @@ class HeaderWithCloseButton extends Component {
                     styles.flexGrow1,
                     styles.justifyContentBetween,
                     styles.overflowHidden,
+                    styles.userSelectNone
                 ]}
                 >
                     {this.props.shouldShowBackButton && (
```

https://user-images.githubusercontent.com/113963320/211602969-d0df1f49-86f1-4dc9-847d-083c526b4f4a.mp4

**Comment by jatinsonijs** [comment]
@tienifr we cannot use user select none as it will create following regression:
https://github.com/Expensify/App/issues/14121

This issue can be fixed by same solution as posted for another issue:
Issue: https://github.com/Expensify/App/issues/13688
Solution: https://github.com/Expensify/App/issues/13688#issuecomment-1363967669

**Comment by mollfpr** [comment]
I agree the text should be selectable.

@jatinsonijs I feel your proposal to complicated. This issue can be fixed with a slight change.

**Comment by jatinsonijs** [comment]
@mollfpr understood your concern, 
we can fix this by just pass userSelectNone to button

```diff
diff --git a/src/components/AvatarCropModal/AvatarCropModal.js b/src/components/AvatarCropModal/AvatarCropModal.js
index 1d9922ead4..a6ad8309c9 100644
--- a/src/components/AvatarCropModal/AvatarCropModal.js
+++ b/src/components/AvatarCropModal/AvatarCropModal.js
@@ -348,7 +348,7 @@ const AvatarCropModal = (props) => {
                                     icon={Expensicons.Rotate}
                                     iconFill={themeColors.inverse}
                                     iconStyles={[styles.mr0]}
-                                    style={[styles.imageCropRotateButton]}
+                                    style={[styles.imageCropRotateButton, styles.userSelectNone]}
                                     onPress={rotateImage}
                                 />
                             </View>
```

**Comment by mollfpr** [comment]
@jatinsonijs Could you explain why the issue happen and why your solution is to fix the issue?

**Comment by jatinsonijs** [comment]
@mollfpr it is default behaviour of browsers whenever we do double click text get selected. With userSelectNone we are preventing this behaviour on particular div.

**Comment by Puneet-here** [comment]
Duplicate of https://github.com/Expensify/App/issues/10515. It was closed because it's not a valuable fix 
cc: @iwiznia

**Comment by mollfpr** [comment]
Thanks @jatinsonijs!

---
>I believe this is because of Icon, as it is using an SVG image it exhibits similar behaviour to text, allowing the user to select on click.

It is only reproducible on the button using an icon. 

>It might be worth fixing this by adding styles.userSelectNone to [Button](https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/Button.js#L205) so we don't see this kind of behaviour again elsewhere.

I agree with this, can you confirm that adding `styles.userSelectNone` to the button will not make a regression? Even though the text on a button is not selectable. Also, could you send a snippet code for this solution? Thanks!

---
@twisterdotcom Do you think this should be closed to like @Puneet-here attached?

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/components/Button.js b/src/components/Button.js
index 6e6fbade77..e7689a7066 100644
--- a/src/components/Button.js
+++ b/src/components/Button.js
@@ -181,7 +181,6 @@ class Button extends Component {
 
         const textComponent = (
             <Text
-                selectable={false}
                 style={[
                     this.props.isLoading && styles.opacity0,
                     styles.pointerEventsNone,
@@ -255,6 +254,7 @@ class Button extends Component {
                 disabled={this.props.isLoading || this.props.isDisabled}
                 style={[
                     this.props.isDisabled ? {...styles.cursorDisabled, ...styles.noSelect} : {},
+                    styles.userSelectNone,
                     styles.buttonContainer,
                     this.props.shouldRemoveRightBorderRadius ? styles.noRightBorderRadius : undefined,
                     this.props.shouldRemoveLeftBorderRadius ? styles.noLeftBorderRadius : undefined,
```

Instead of adding the `styles.userSelectNone` to only the current effected button (@jatinsonijs) or to only the icon (@Ollyws). I think it may be better to apply the style in the `<Pressable />` directly and avoid multiple styles declaration.

**Comment by mollfpr** [comment]
@s77rt Removing `selectable={false}` will make the text on the web selectable because the default value is `true` on RNW.

**Comment by jatinsonijs** [comment]
@mollfpr we can do it at low level like in Button as suggested by @Ollyws,but its only happen on double press I haven't any other area where we are going to do double tap.

**Comment by mollfpr** [comment]
@jatinsonijs Yes, but since it's a reusable component we can prevent the same issue occur in the future. Toggled button with an icon will not only be for the avatar modal but could be anywhere.

**Comment by s77rt** [comment]
@mollfpr No, it wont.
The text is wrapped inside the button (`<Pressable />`) and we already passed `styles.userSelectNone` there.
So the `selectable={false}` is actually redundant.



https://user-images.githubusercontent.com/16493223/211837511-ac9b20c7-96ee-4c7b-9f91-0774b49c4d76.mp4

**Comment by getusha** [comment]
this issue is not reproducible in prod. right?

**Comment by mollfpr** [comment]
I can reproduce it on production with Safari.

**Comment by getusha** [comment]
### Proposal

```diff
@@ -349,6 +349,9 @@ const AvatarCropModal = (props) => {
                                     iconFill={themeColors.inverse}
                                     iconStyles={[styles.mr0]}
                                     style={[styles.imageCropRotateButton]}
+                                    onMouseDown={(e) => {
+                                        if (getSelection().isCollapsed) e.preventDefault()
+                                    }}
                                     onPress={rotateImage}
                                 />
                             </View>
```

### Result

https://user-images.githubusercontent.com/75031127/211843091-96ed3c33-3eb9-41b0-97d1-42c9aabd7c0e.mov

checking `getSelection().isCollapsed` is useful because if we have selected text we still want to dismiss it
we're preventing this issue from happening

https://user-images.githubusercontent.com/75031127/211843823-7d1c6582-a02c-4258-a5f8-5868eace6794.mov

cc @mollfpr

**Comment by mollfpr** [comment]
Thank you for the proposals! Let's wait for clarity on this https://github.com/Expensify/App/issues/14139#issuecomment-1378739288

**Comment by s77rt** [comment]
@0xmiroslav Basically every component that uses `<Pressable />` is effected including `<Button />` and `<Checkbox />`.
We can apply the same fix for checkbox.

**Comment by s77rt** [comment]
@parasharrajat That's looks cool! Can you check if using `<button>` instead of `<div>` works fine in our app? I think we have no `<button>` rendered at all (RNW does not render any buttons I suppose)

**Comment by s77rt** [comment]
@0xmiroslav I don't think so, the solution is only for buttons, everything else will stay the same

**Comment by s77rt** [comment]
Just tested @parasharrajat solution. The button works as expected and it's rendered as `<button />` but it's not correct. The button contains `<div />`s and that should not happen. `<button />` may only contain phrasing content and `<div />` is not.

[source1](https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element)
[source2](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2)

___

You can test here
https://validator.w3.org/nu/#textarea

This is **OK**
```
<!DOCTYPE html>
<html lang="en">
<head>
<title>Page Title</title>
</head>
<body>

<button>test</button>

</body>
</html>
```


This is **NOT OK**
```
<!DOCTYPE html>
<html lang="en">
<head>
<title>Page Title</title>
</head>
<body>

<button><div>test</div></button>

</body>
</html>
```

**Comment by s77rt** [comment]
I think the best option here is to disable selection for buttons. Although disabling selection may not be appealing this is the best we can do I think.

**Comment by mollfpr** [comment]
The party is here https://expensify.slack.com/archives/C01GTK53T8Q/p1673539746750589

**Comment by getusha** [comment]
```diff index 6e6fbade77..d31f1702f4 100644
--- a/src/components/Button.js
+++ b/src/components/Button.js
@@ -251,7 +251,10 @@ class Button extends Component {
                 }}
                 onPressIn={this.props.onPressIn}
                 onPressOut={this.props.onPressOut}
-                onMouseDown={this.props.onMouseDown}
+                onMouseDown={(e)=>{
+                    e.stopPropagation();
+                    this.props.onMouseDown}
+                }
                 disabled={this.props.isLoading || this.props.isDisabled}
                 style={[
                     this.props.isDisabled ? {...styles.cursorDisabled, ...styles.noSelect} : {},
```

the selection sort of blinks but solves the issue

**Comment by getusha** [comment]
`event.stopPropagation()` is a method to prevent the event from bubbling up to the parent element.

**Comment by mollfpr** [comment]
@getusha The result doesn't look good and does not meet the expected result.
https://user-images.githubusercontent.com/25520267/212134824-8b365999-f874-4c6a-84df-0c0927ba2239.mov

**Comment by getusha** [comment]
@mollfpr you're right but the text is not remaining highlighted right?

**Comment by mollfpr** [comment]
@getusha Yes, but still not the result we looking for.

**Comment by getusha** [comment]
@mollfpr i believe this should fix the issue effectively

```diff --git a/src/components/Button.js b/src/components/Button.js
index 6e6fbade77..c73df6cf03 100644
--- a/src/components/Button.js
+++ b/src/components/Button.js
@@ -251,7 +251,10 @@ class Button extends Component {
                 }}
                 onPressIn={this.props.onPressIn}
                 onPressOut={this.props.onPressOut}
-                onMouseDown={this.props.onMouseDown}
+                onMouseDown={(e)=>{
+                    getSelection().isCollapsed && e.preventDefault();
+                    this.props.onMouseDown();
+                }}
                 disabled={this.props.isLoading || this.props.isDisabled}
                 style={[
                     this.props.isDisabled ? {...styles.cursorDisabled, ...styles.noSelect} : {},
```

**Comment by mollfpr** [comment]
@getusha Thanks, I’ll review it later.

**Comment by mollfpr** [comment]
@danieldoglas @twisterdotcom We still need confirmation about this issue.

The other thing is I'm inclined toward what @parasharrajat [comment](https://github.com/Expensify/App/issues/14139#issuecomment-1379475849). The chance is will fix other similar issues.

>IMO, The best solution for it will be to render the button element instead of div. I am tracking an issue for this https://github.com/necolas/react-native-web/issues/1899 on RN-web which should be released IMO or we can patch our >fork if that works. It should fix all such selection issues. There is a PR doing that https://github.com/necolas/react-native->web/pull/1944 we can do that same on our fork. It is a small change.

But this will lead to another issue @s77rt mention [here](https://github.com/Expensify/App/issues/14139#issuecomment-1379553500).

**Comment by danieldoglas** [comment]
I agree that we don't want to create something that is semantically incorrect. Is this worthy for us creating a PR on react-native that would solve this semantically?

**Comment by danieldoglas** [comment]
@twisterdotcom how big do you feel this issue actually is? I think that would determine how we proceed on the discussion here.

**Comment by s77rt** [comment]
@parasharrajat It is semantically incorrect, although I didn't notice any major bug, it's still not the correct thing to do as it may lead to future bugs or browsers not doing certain things where they should.

Is disabling the selection on buttons also a way for future bugs too? It looks safer

**Comment by danieldoglas** [comment]
Since this is in the pipeline for solving in react-native, and we don't actually think this is an urgent bug IMO we should hold this on https://github.com/necolas/react-native-web/issues/1899.

**Comment by danieldoglas** [comment]
Adding HOLD on this.

**Comment by danieldoglas** [comment]
Removed `External` and `Help Wanted`

**Comment by twisterdotcom** [comment]
Still waiting on https://github.com/necolas/react-native-web/issues/1899

**Comment by s77rt** [comment]
Looks like the linked issue is resolved (https://github.com/necolas/react-native-web/pull/1944 has been merged)
I think we should now test the changes? (and update our RNW fork)

**Comment by roryabraham** [comment]
Putting this on HOLD for https://github.com/Expensify/App/issues/16660 – @parasharrajat feel free to comment in there and I'll co-assign you.

**Comment by MahmudjonToraqulov** [comment]
### Proposal

### Please re-state the problem that we are trying to solve in this issue.
Web - Profile - The text on the modal Drag, zoom, and rotate your image to your preferred is highlighted when rotate the picture the picture

### What is the root cause of that problem?
The texts above should have been added "user-select: none;" for not taking this problem.

### What changes do you think we should make in order to solve the problem?
I found ideal solution. We just need to add this style to the text elements to solve this issue:
`user-select: none;`

### What alternative solutions did you explore? (Optional)
Not yet

### Result

[screen-capture.webm](https://user-images.githubusercontent.com/91727683/230762433-e6260a81-7500-4f59-a0eb-8a5fd282d598.webm)

**Comment by MelvinBot** [comment]
📣 @MahmudjonToraqulov! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MahmudjonToraqulov** [comment]
Contributor details
Your Expensify account email: toraqulovmahmudjon@gmail.com
Upwork Profile Link:  https://www.upwork.com/freelancers/~01347b2040312758ec

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by MahmudjonToraqulov** [comment]
@parasharrajat
Sorry. I'm new here. I knew that when GH has `External` or `Help Wanted` badge than It's open to proposal. Am I right?

**Comment by MahmudjonToraqulov** [comment]
@parasharrajat Ok. Thanks for the update.

**Comment by MelvinBot** [comment]
@danieldoglas, @twisterdotcom, @mollfpr Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by danieldoglas** [comment]
still on hold

**Comment by danieldoglas** [comment]
moving this to weekly while is on hold

**Comment by twisterdotcom** [comment]
Let's go `Monthly`.

**Comment by danieldoglas** [comment]
Still on hold. still on hold

**Comment by danieldoglas** [comment]
Following the new guidelines for external issues, removing myself from this until we have proposals.

**Comment by twisterdotcom** [comment]
Still held.

**Comment by twisterdotcom** [comment]
Still HELD

**Comment by twisterdotcom** [comment]
Held, but not for long!

**Comment by twisterdotcom** [comment]
@mollfpr can we unhold this, or do we also need to wait for #31381?

**Comment by twisterdotcom** [comment]
Still held on https://github.com/Expensify/App/issues/31381

**Comment by roryabraham** [comment]
Looks like this can come off HOLD

**Comment by mollfpr** [comment]
I'll retest the issue 👍

**Comment by mollfpr** [comment]
I think this issue was already fixed!

https://github.com/Expensify/App/assets/25520267/7b9fd809-dcfa-4df3-ba7f-6189627c4e8f

Although, the highlight remains, but I think it's expected. I do the same with a new project; it keeps the highlight on the text after pressing a button.

https://codesandbox.io/p/sandbox/stupefied-matan-2y8w2z
